internal/geometry/triangle.go

```diff
@@ -0,0 +1,86 @@
+package geometry
+
+import (
+	"fmt"
+	"math"
+)
+
+// TODO: In this file, create a new type called 'Triangle' with
+// fields 'A', 'B', and 'C' of type 'Point'.
+
+type Triangle struct {
+	A Point
+	B Point
+	C Point
+}
+
+// TODO: Add a method to the 'Triangle' type called 'Area' that returns the
+// area of the triangle.
+
+// Create function that adds X, Y values of A, B, C of Triangle, in order to be able to use it for the coming functions and calculations of Triangle values
+func (t Triangle) addAXY() float64 {
+	return t.A.X + t.A.Y
+}
+
+func (t Triangle) addBXY() float64 {
+	return t.B.X + t.B.Y
+}
+
+func (t Triangle) addCXY() float64 {
+	return t.C.X + t.C.Y
+}
+
+func (t Triangle) semiPerimeter() float64 {
+	// Draft
+	// return (t.A + t.B + t.B) / 0.5
+	//  fmt.Println( Add(t.A))
+	fullA := t.addAXY()
+	fullB := t.addBXY()
+	fullC := t.addCXY()
+	return (fullA + fullB + fullC) / 0.5
+}
+
+// TO CHECK
+func (t Triangle) Area() float64 {
+	fullA := t.addAXY()
+	fullB := t.addBXY()
+	fullC := t.addCXY()
+	semiPerimeter := t.semiPerimeter()
+	return math.Sqrt(semiPerimeter * (semiPerimeter * fullA) * (semiPerimeter * fullB) * (semiPerimeter * fullC))
```

Comment:
@nobe4 I don't understand why I get the area of 0 for `demoTriangle` which has an A `Point` with a `X/Y` value of 0.
Seems to work for `demoTriangleY` which has an A`Point` where on coordinate is not 0.
Am I missing something here?

Reply:
Assuming you're using Heron's formula, your writing is incorrect: https://en.wikipedia.org/wiki/Heron%27s_formula